mrp_subcontracting_inhibit/models/purchase_order.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2022 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    subcontracting_inhibit = fields.Boolean(string="Inhibit subcontracting")
+
+    @api.model
+    def _prepare_purchase_order_line_from_procurement(
+        self, product_id, product_qty, product_uom, company_id, values, po
+    ):
+        """We need to inject the context to set the right price"""
+        if values.get("route_ids"):
+            subcontracting_inhibit_value = any(
```

Comment:
When you call super methods, you should always protect possible modifications on the "evaled" variables, so give this default:

```suggestion
        subcontracting_inhibit_value = False
        if values.get("route_ids"):
            subcontracting_inhibit_value = any(
```